luaui/Widgets/gui_options.lua

```diff
@@ -2890,16 +2890,19 @@ function init()
 			onchange = function(_, value)
 				local keyFile = keyLayouts.keybindingLayoutFiles[value]
 
-				if not keyFile or keyFile == '' or not VFS.FileExists(keyFile) then
+				if not keyFile or keyFile == '' then
 					return
 				end
 
-				Spring.SetConfigString("KeybindingFile", keyFile)
+				local isCustom = keyLayouts.keybindingPresets["Custom"] == keyFile
 
-				if WG['bar_hotkeys'] and WG['bar_hotkeys'].reloadBindings then
-					WG['bar_hotkeys'].reloadBindings()
+				if isCustom and not VFS.FileExists(keyFile) then
+					Spring.SendCommands("keysave " .. keyFile)
```

Comment:
Why do we `keysave` here?

Reply:
The reason is to generate a first-time uikeys.txt file for a new user based on their current keybinds.

Reply:
kk, seems reasonable.

Reply:
kk, seems reasonable